tests/test_infer.py

```diff
@@ -172,35 +174,24 @@ def test_calculator_distributions_without_teststatistic(qtilde):
         (9, 1.1285884059538408e-19),
     ],
 )
-@pytest.mark.parametrize(
-    "backend,rtol",
-    [
-        (pyhf.tensor.numpy_backend(), 1e-8),
-        (pyhf.tensor.tensorflow_backend(), 1e-5),
-        (pyhf.tensor.pytorch_backend(), 1e-5),
-        (pyhf.tensor.jax_backend(), 1e-8),
-    ],
-)
-def test_asymptotic_dist_low_pvalues(backend, rtol, nsigma, expected_pval):
-    atol = 0
-    pyhf.set_backend(backend)
+def test_asymptotic_dist_low_pvalues(backend, nsigma, expected_pval):
+    rtol = 1e-8
+    if hasattr(backend[0], "dtypemap") and (
+        backend[0].dtypemap["float"] == torch.float32
+        or backend[0].dtypemap["float"] == tf.float32
+    ):
```

Comment:
`if backend.name in ["pytorch", "tensorflow"]` is easier to check for fyi (and less imports)

Reply:
mhh my idea was to check for the dtypes since the achievable precision depends on this. There is for example also a case for pytorch with 64bits in the backend fixture
https://github.com/scikit-hep/pyhf/blob/cb4d37ba89e2aa9b43f6db3e168841b97d7079e6/tests/conftest.py#L82

Reply:
alternatively we should add a dtypemap for all backends (i'm surprised we didn't add this)

Reply:
there is a mapping for all backends, but only torch and tf have it as an argument in the constructor, numpy and jax have it as an argument for `astensor`. I agree it might make sense to unify it and have a single argument/property saying what precision it should use. Should we do it in this PR or a separate one?

Reply:
i can do it in an upcoming PR. opening an issue

Reply:
opened up #910 to deal with this for @nikoladze . We'll merge that in asap.